src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -47,90 +51,107 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
     [accountSpecifiers],
   )
 
+  /**
+   * tx history unsubscribe and cleanup logic
+   */
+  useEffect(() => {
+    // account specifiers changing will trigger this effect
+    // we've disconnected/switched a wallet, unsubscribe from tx history and clear tx history
+    if (txIds.length) {
```

Comment:
why does `txIds.length` matter to whether we unsubscribe or not? If I have a wallet with no transactions and then switch wallets, we should still unsubscribe.

Reply:
updated, please check